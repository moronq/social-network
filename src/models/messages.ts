export type ChateMessageType = {
  message: string
  photo: string
  userId: number
  userName: string
}
