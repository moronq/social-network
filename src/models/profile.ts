export type ProfileTypeResponse = {
  aboutMe: string
  userId: number | null
  lookingForAJob: boolean
  lookingForAJobDescription: string
  fullName: string
  contacts: ContactsType
  photos: PhotosType
}

export type PhotosType = {
  small: string
  large: string
}

export type ProfileTypeStatus = {
  status: string
}

export type ProfileType = ProfileTypeResponse & { error: string }

type ContactsType = {
  github: string | null
  vk: string | null
  facebook: string | null
  instagram: string | null
  twitter: string | null
  website: string | null
  youtube: string | null
  mainLkink: string | null
}

export type ProfileResponse<T = any> = {
  resultCode: number
  data: T
  fieldsError: Array<string>
  messages: Array<string>
}
