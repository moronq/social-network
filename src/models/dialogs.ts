import { PhotosType } from './profile'

export type dialogType = {
  id: number
  userName: string
  hasNewMessages: boolean
  lastDialogActivityDate: string
  lastUserActivityDate: string
  newMessagesCount: number
  photos: PhotosType
}
