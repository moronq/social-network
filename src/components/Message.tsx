import React, { FC } from 'react'
import { ChateMessageType } from '../models/messages'
import avatar from './../assets/img/no-avatar-small.png'

const Message: FC<{ message: ChateMessageType }> = ({ message }) => {
  const userPhoto = message.photo ?? avatar

  return (
    <li className="flex gap-5">
      <img src={userPhoto} alt="avatar" className="w-12 h-12 rounded-full " />
      <div>
        <p>{message.userName}</p>
        <span>{message.message}</span>
      </div>
    </li>
  )
}

export default Message
