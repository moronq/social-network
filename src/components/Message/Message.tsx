import React, { FC } from 'react'
import { Link } from 'react-router-dom'
import { ChateMessageType } from '../../models/messages'
import avatar from './../../assets/img/no-avatar-small.png'

type PropsType = {
  message: ChateMessageType
}

const Message: FC<PropsType> = ({ message }) => {
  const userPhoto = message.photo ?? avatar

  return (
    <li className="flex gap-5">
      <Link to={`/profile/${message.userId}`}>
        <img src={userPhoto} alt="avatar" className="w-12 h-12 rounded-full " />
      </Link>
      <div>
        <Link to={`/profile/${message.userId}`}>
          <p>{message.userName}</p>
        </Link>
        <span>{message.message}</span>
      </div>
    </li>
  )
}

export default Message
