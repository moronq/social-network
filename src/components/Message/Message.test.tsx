import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { ChateMessageType } from '../../models/messages'
import Message from './Message'

const mockMessage: ChateMessageType = {
  message: 'test message',
  photo:
    'https://social-network.samuraijs.com/activecontent/images/users/19481/user-small.jpg?v=7',
  userId: 19481,
  userName: 'dmitriy199427',
}

describe('Message test group', () => {
  test('Message renders without error', () => {
    render(
      <MemoryRouter>
        <Message message={mockMessage} />
      </MemoryRouter>
    )
    expect(screen.getByText('test message')).toBeInTheDocument
    expect(screen.getByText('dmitriy199427')).toBeInTheDocument
  })

  test('avatar check', () => {
    render(
      <MemoryRouter>
        <Message message={mockMessage} />
      </MemoryRouter>
    )
    const img = screen.getByRole('img') as HTMLImageElement
    expect(img.src).toContain(
      'https://social-network.samuraijs.com/activecontent/images/users/19481/user-small.jpg?v=7'
    )
    expect(img.alt).toContain('avatar')
  })

  test('snapshot Message', () => {
    const message = render(
      <MemoryRouter>
        <Message message={mockMessage} />
      </MemoryRouter>
    )
    expect(message).toMatchSnapshot()
  })
})
