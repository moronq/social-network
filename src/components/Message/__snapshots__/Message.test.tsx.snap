// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message test group snapshot Message 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="flex gap-5"
      >
        <a
          href="/profile/19481"
        >
          <img
            alt="avatar"
            class="w-12 h-12 rounded-full "
            src="https://social-network.samuraijs.com/activecontent/images/users/19481/user-small.jpg?v=7"
          />
        </a>
        <div>
          <a
            href="/profile/19481"
          >
            <p>
              dmitriy199427
            </p>
          </a>
          <span>
            test message
          </span>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="flex gap-5"
    >
      <a
        href="/profile/19481"
      >
        <img
          alt="avatar"
          class="w-12 h-12 rounded-full "
          src="https://social-network.samuraijs.com/activecontent/images/users/19481/user-small.jpg?v=7"
        />
      </a>
      <div>
        <a
          href="/profile/19481"
        >
          <p>
            dmitriy199427
          </p>
        </a>
        <span>
          test message
        </span>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
